### PR TITLE
[train] Rename `train(config)` to `train_fn(config)`

### LINKED ISSUE #38493: Release test tune_cloud_durable_upload.aws failed
Release test **tune_cloud_durable_upload.aws** failed. See https://buildkite.com/ray-project/release-tests-branch/builds/2051#0189fd8d-665a-4ac3-80fb-820482e63380 for more details.

Managed by OSS Test Policy

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

Our API change from `tune.report()` to `train.report()` can lead to namespace clashes when the training function is called `train`.

This has been an issue in many test migrations, including #39050 and the current failure of #38493.

This PR does a global replace of all training function defined as `def train(config)` with `def train_fn(config)` to avoid future clashes.

## Related issue number

Closes #38493
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by matthewdeng — APPROVED** [review]
Looks good assuming tests pass. Let's pick this into 2.7

**Review by sven1977 — APPROVED** [review]
Thanks @krfricke

File: doc/source/rllib/rllib-advanced-api.rst
[the remark this answers is not in the document]
Comment (krfricke) [reply doc/source/rllib/rllib-advanced-api.rst:1]: Ah I see. Yes let's make it consistent, but I think there are plenty of (other) places where we have `train_fn` already. Maybe do this in one go on master after this is merged.

**Comment by krfricke** [comment]
Release test passed here: https://buildkite.com/ray-project/release-tests-pr/builds/51367

**Pull request merged**